### PR TITLE
projects: Add cobot enhancement project

### PR DESCRIPTION
**Pull request by meetmangukiya**
**For short term contributors:** we understand that getting your commits well
defined like we require is a hard task and takes some learning. If you
look to help without wanting to contribute long term there's no need
for you to learn this. Just drop us a message and we'll take care of brushing
up your stuff for merge!

### Checklist

- [ ] I read the [commit guidelines](http://coala.io/commit) and I've followed
      them.
- [ ] I ran coala over my code locally. (*All commits have to pass
      individually.* It is not sufficient to have "fixup commits" on your PR,
      our bot will still report the issues for the previous commit.) You will
      likely receive a lot of bot comments and build failures if coala does not
      pass on every single commit!

After you submit your pull request, **DO NOT click the 'Update Branch' button.**
When asked for a rebase, consult [coala.io/rebase](https://coala.io/rebase)
instead.

Please consider helping us by reviewing other peoples pull requests as well:

- pick up any PR at <https://coala.io/review>
- review it (check <https://coala.io/reviewing> for more info)
- if you are sure that it needs work, use `cobot mark wip <URL>` to get it out
  of the review queue.

The more you review, the more your score will grow at coala.io and we will
review your PRs faster!

### CODE REVIEW

**Comment by Udayan12167** [comment]
So you have to add references to your issues per milestone. A link to the repository. A link to a PDF of your proposal. And an in progress tag.

**Comment by meetmangukiya** [comment]
@Udayan12167 added the issues in the `issues` key. Added the link to the repository and the proposal pdf in the description right before the milestones since there was no key in the projects.js that matched these

File: data/projects/cobot_enhancement.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+many people don't seem to respect/ignore knowingly or unknowingly not making it
+worthwile having good up-to-date docs. We aim at using topic modeling i.e. take
+help of machine learning and natural language processing(NLP) for this one.
+
+Repository: https://github.com/coala/corobo
+Proposal PDF: https://meetmangukiya.github.io/GSoC_cobot_enhancement.pdf
+
+#### Milestones
+
+##### PREPARATION/BONDING
```
Comment (Udayan12167) [inline data/projects/cobot_enhancement.md]: Add milestone links to this?

File: data/projects/cobot_enhancement.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+* The applicant has made a cEP of how the new plugins will be implemented, how
+  will it be tested, etc.
+
+##### CODING PHASE 1
```
Comment (Udayan12167) [inline data/projects/cobot_enhancement.md]: Same

File: data/projects/cobot_enhancement.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+* Port all the plugins to errbot.
+* Write half of the new features proposed.
+* Write the test helper and all the merged plugins should have unit tests.
+
+##### CODING PHASE 2
```
Comment (Udayan12167) [inline data/projects/cobot_enhancement.md]: Same

File: data/projects/cobot_enhancement.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+* Write the other half of the new features proposed.
+* Start implementing the topic modeling functions, etc.
+
+##### CODING PHASE 3
```
Comment (Udayan12167) [inline data/projects/cobot_enhancement.md]: Same

File: data/projects.js [hunk 5]
```diff
@@ -536,7 +536,7 @@ coala should support generating metrics for your code.",
 		"mentors" : ["Adrianzatreanu", "Udayan12167", "fneu"],
 		"initiatives" : ["GSoC", "RGSoC"],
 		"tags" : ["Bears", "cib"],
-		"markdown": "improve_installation.md",
+		"mar:kdown": "improve_installation.md",
```
Comment (Udayan12167) [inline data/projects.js]: Why this change?

**Comment by jayvdb** [comment]
@meetmangukiya those messages above ... they look like a bug in InvalidLinkBear .. ?  Has the bug been raised?
We should work around that with an ignore in the .coafile ?

File: data/projects/cobot_enhancement.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
+This project is about cobot, the automation chat bot that is used at coala.
+Currently cobot is used for several things:
+- Assign issues
```
Comment (jayvdb) [inline data/projects/cobot_enhancement.md]: Please create nice markdown, with whitespace where appropriate.  So reviewers can read it.  Use our bear to fix it.

**Comment by meetmangukiya** [comment]
@jayvdb no, the issue hasn't been raised according a quick search with "invalid" keyword in the issues of bears. Raising one: https://github.com/coala/coala-bears/issues/1780

File: data/projects/cobot_enhancement.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+3. Working functionality to get answers from the docs and summarize the doc to
+   answer in concise manner.
+
+In week 11 and 12:
+1. Based on the scores determine if the question can be answered from our docs,
+   if not just return a google result for the same question.
+2. Buffer.
+3. Documentation(particularly about the algorithm used for retrieving the
+   answers).
+
```
Comment (jayvdb) [inline data/projects/cobot_enhancement.md]: trailing blank line.

File: data/projects/cobot_enhancement.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+2. Start working on the passage retrieval part to _get_ the answers from the
+   documentation.
+3. Working functionality to get answers from the docs and summarize the doc to
+   answer in concise manner.
+
+In week 11 and 12:
+1. Based on the scores determine if the question can be answered from our docs,
+   if not just return a google result for the same question.
+2. Buffer.
+3. Documentation(particularly about the algorithm used for retrieving the
```
Comment (jayvdb) [inline data/projects/cobot_enhancement.md]: space before `(`

File: data/projects/cobot_enhancement.md [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+In week 1 and 2: All the existing scripts in gitlab/coala/cobot should be
+ported to github/coala/corobo. All the plugins should be tested with 100%
+coverage.
+
+In week 3 and 4: At least two new plugins from
+https://github.com/coala/corobo/milestone/5 should be written and tested.
+
+##### CODING PHASE 2
+
+In week 5 and 6:
```
Comment (jayvdb) [inline data/projects/cobot_enhancement.md]: I think you need a blank line after this, to be proper markdown syntax.

File: data/projects/cobot_enhancement.md (hunk 7 above)
Comment (meetmangukiya) [inline data/projects/cobot_enhancement.md]: I think something's wrong with my editor. How can I add trailing blank line in every file?! 

Sorry. (emberresing)

Comment (jayvdb) [inline data/projects/cobot_enhancement.md]: That is your problem, with your editor / configuration.
You should run linters and read `git show` before submitting to ensure you've fix it.

**Comment by jayvdb** [comment]
@rultor merge

**Comment by jayvdb** [comment]
reack 22be121

**Comment by jayvdb** [comment]
ack 22be121
